src/imars3d/backend/io/metadata.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+"""
+Metadata class for IMars3D
+"""
+import numpy as np
+import param
+import tifffile
+from pathlib import Path
+
+
+# setup module level logger
+logger = param.get_logger(__name__)
+logger.name = __name__
+
+
+class MetaData(param.Parameterized):
+    """
+    Metadata extracted from given file
+    """
+
+    filename = param.Filename(doc="full path to the file.")
+    datatype = param.Selector(
+        default="ct",
+        objects=[
+            "radiograph",
+            "ct",
+            "openbeam",
+            "ob",
+            "darkcurrent",
+            "dc",
+        ],
+        doc="data type [ct|ob|dc]",
+    )
+    metadata_index = param.List(
```

Comment:
This looks like a large variety of changes to the underlying file are going to break it. Is that right?

Also, since these values are repeated in the tests, it would be good to make them file-level constants so they can be shared in both places.

Reply:
The answer to this is slightly more complicated.
If the client do not specify the index of the metadata by setting/overwriting metadata_index, it will error our when the data file switch to a different set of indices (probably a change at DAS level).
However, if the client properly sets the metadata_index when intilizating metadata object and do not set the data type (as it will trigger the update procedure), it should be fine.

I am repeating the info in tests as I am trying to avoid setting class level constants.

Reply:
What about a file-level `dict` with the values. Undocumented magic numbers are a big pain, but the versions in the source are fairly well documented. The ones in the test aren't.

Reply:
it wouldn't be a `dict`, but a list. Will that be okay?